### PR TITLE
Update feed cookie

### PR DESCRIPTION
**Pull request by kumamidori**
In my operation environment, the server returns many unnecessary cookies data per request. 
the result looks like this:

 `JsonTransportFeed` class:

```
        if ($res->hasHeader('Set-Cookie')) {
            // ↓ Add temporary dump for cookie
            $cookiesSrc = $res->getHeaders()['Set-Cookie'];
            $json = json_encode($cookiesSrc);
```

`$json`:

```
[
  "rur=VLL; Domain=.instagram.com; HttpOnly; Path=\/; Secure",
  "mid=xxxxxxxx; Domain=.instagram.com; expires=Sun, 10-Jun-2029 09:02:47 GMT; Max-Age=315360000; Path=\/; Secure",
  "urlgen=\"{\\\"114.19.21.81\\\": 2516}:xxxxx:xxxxxx\"; Domain=.instagram.com; HttpOnly; Path=\/; Secure",
  "csrftoken=xxxxxxx; Domain=.instagram.com; expires=Thu, 11-Jun-2020 09:02:47 GMT; Max-Age=31449600; Path=\/; Secure",
  "__to_be_deleted__instagram.com=\"\"; Domain=instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; Domain=instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; Domain=.instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; Domain=i.instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; Domain=.i.instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; Domain=www.instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; Domain=.www.instagram.com; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
  "3=\"\"; expires=Thu, 01-Jan-1970 00:00:00 GMT; Max-Age=0; Path=\/",
   // ... 
]
```

The cookie header size increases per pagination.
Too many cookie data will cause the curl error at last.

### CODE REVIEW

File: src/Instagram/Transport/JsonTransportFeed.php [hunk 1]
```diff
@@ -85,9 +88,16 @@ public function fetchData($userName, $limit)
...
         $newCache->setUserId($cache->getUserId());
         if ($res->hasHeader('Set-Cookie')) {
+            $saveCookies = [];
+            foreach ($res->getHeaders()['Set-Cookie'] as $cookie) {
+                $setCookie = SetCookie::fromString($cookie);
+                if (in_array($setCookie->getName(), $this->requiresCookieKeys, true)) {
+                    $saveCookies[] = $cookie;
+                }
+            }
             $newCache->setCookie($res->getHeaders()['Set-Cookie']);
```
Comment (pgrimaud) [inline src/Instagram/Transport/JsonTransportFeed.php]: I guess you should remove this line?

File: src/Instagram/Transport/JsonTransportFeed.php [hunk 2]
```diff
@@ -15,6 +16,8 @@ class JsonTransportFeed extends TransportFeed
      */
     private $endCursor;
 
+    private $requiresCookieKeys = ['rur', 'mid', 'mcd', 'urlgen', 'csrftoken'];
```
Comment (pgrimaud) [inline src/Instagram/Transport/JsonTransportFeed.php]: `const REQUIRES_COOKIES_KEYS = [...]` is better here

File: src/Instagram/Transport/JsonTransportFeed.php (hunk 1 above)
Comment (kumamidori) [reply src/Instagram/Transport/JsonTransportFeed.php]: Thank you for your review.
I fixed the issue commented by you. ↓ I used git rebase in practice. 

```diff
 src/Instagram/Transport/JsonTransportFeed.php | 3 +--                                
 1 file changed, 1 insertion(+), 2 deletions(-)               
                                                                                      
diff --git a/src/Instagram/Transport/JsonTransportFeed.php b/src/Instagram/Transport/J
index bb6401d..293757d 100644                                                         
--- a/src/Instagram/Transport/JsonTransportFeed.php                                   
+++ b/src/Instagram/Transport/JsonTransportFeed.php                                   
@@ -95,9 +95,8 @@ class JsonTransportFeed extends TransportFeed                       
                     $saveCookies[] = $cookie;                             
                 }                                                          
             }                                                                        
-            $newCache->setCookie($res->getHeaders()['Set-Cookie']);                  
+            $newCache->setCookie($saveCookies);                                      
         }                                                                            
-        $newCache->setCookie($saveCookies);                                          
         $this->cacheManager->set($newCache, $userName);                 
                                                                               
         return $data->data->user;                         
```

File: src/Instagram/Transport/JsonTransportFeed.php (hunk 2 above)
Comment (kumamidori) [reply src/Instagram/Transport/JsonTransportFeed.php]: Thank you. I fixed the issue.  ↓ I used git rebase in practice.

```diff
--- a/src/Instagram/Transport/JsonTransportFeed.php
+++ b/src/Instagram/Transport/JsonTransportFeed.php
@@ -16,7 +16,7 @@ class JsonTransportFeed extends TransportFeed
      */
 
-    private $requiresCookieKeys = ['rur', 'mid', 'mcd', 'urlgen', 'csrftoken'];
+    const REQUIRES_COOKIES_KEYS = ['rur', 'mid', 'mcd', 'urlgen', 'csrftoken'];
 
     /**
      * JsonTransportFeed constructor.
@@ -87,15 +87,14 @@ class JsonTransportFeed extends TransportFeed
         // save to cache for next request
         $newCache = new Cache();
         $newCache->setUserId($cache->getUserId());
        if ($res->hasHeader('Set-Cookie')) {
            $saveCookies = [];
             foreach ($res->getHeaders()['Set-Cookie'] as $cookie) {
                 $setCookie = SetCookie::fromString($cookie);
-                if (in_array($setCookie->getName(), $this->requiresCookieKeys, true)) {
+                if (in_array($setCookie->getName(), self::REQUIRES_COOKIES_KEYS, true)) {
                     $saveCookies[] = $cookie;
```

File: src/Instagram/Transport/JsonTransportFeed.php (hunk 1 above)
Comment (pgrimaud) [reply src/Instagram/Transport/JsonTransportFeed.php]: 👍

File: src/Instagram/Transport/JsonTransportFeed.php (hunk 2 above)
Comment (pgrimaud) [reply src/Instagram/Transport/JsonTransportFeed.php]: 👍

**Comment by pgrimaud** [comment]
Thanks @kumamidori

**Comment by kumamidori** [comment]
Thanks @pgrimaud  🌷

**Pull request merged**